src/osp/CoordinateSpaces/CartesianSimple.cpp

```diff
@@ -47,36 +47,30 @@ void CoordspaceCartesianSimple::update_exchange(
         rData.m_velocities.push_back(vel);
     }
 
-    rSpace.m_toAdd.clear();
```

Comment:
does m_toAdd get cleaned up by something external to this function?

My reason for suggesting the use of std::exchange() was to limit the lifetime of the data held by m_toAdd.

But if the data is going to be cleaned up soon after, it's fine to not immediately clear it.

Reply:
These two lines in [simple.cpp line 133](https://github.com/TheOpenSpaceProgram/osp-magnum/blob/884376500efb7162531c28e609d7c890dc41e155/src/test_application/universes/simple.cpp#L133) act like two different tasks that depend on m_toAdd.
```cpp
rUni.coordspace_update_sats(coordIndex);
CoordspaceCartesianSimple::update_exchange(rUni, rSpace, *pData);
```

 It would be odd if one cleared the queue, and the other didn't, as they don't modify the same data and can be run in parallel eventually. This is soon intended to be part of the 'universe update function' to be added very soon.